tool/lib/test/unit/testcase.rb

```diff
@@ -50,6 +50,23 @@ def mingw? platform = RUBY_PLATFORM
         /mingw/ =~ platform
       end
 
+      def main_ractor?
+        return true if !defined?(Ractor)
+        Ractor.current == Ractor.main
+      end
+
+      def non_main_ractor?
+        not main_ractor?
+      end
+
+      def multiple_ractors?
+        ENV["RUBY_TESTS_WITH_RACTORS"].to_i > 1
```

Comment:
cache the value or define different methods with the envval?
some tests can remove ENVVAL.